src/frontend/Driver.ml

```diff
@@ -51,12 +54,68 @@ let print_ident (ident : Ident.t CS.node) : command =
   | _ ->
     RM.throw @@ Err.RefineError (Err.UnboundVariable ident.node, ident.info)
 
+let protect m =
+  RM.trap m |>> function
+  | Ok return ->
+    RM.ret @@ Ok return
+  | Error (Err.RefineError (err, info)) ->
+    let+ () = RM.emit ~lvl:`Error info RefineError.pp err in
+    Error ()
+  | Error exn ->
+    let* env = RM.read in
+    let+ () = RM.emit ~lvl:`Error (RefineEnv.location env) PpExn.pp exn in
+    Error ()
+
+(* Interfaces and Imports *)
+
+type iface = { contents : ST.t;
+               hash: Digest.t }
+
+(* Try to load an interface file, failing if the file does not exist or is out of date. *)
+let load_iface_opt path =
+  let src_path = path ^ ".cooltt" in
+  let iface_path = path ^ ".cooltti" in
+  if Sys.file_exists iface_path then
+    let chan = open_in_bin iface_path in
+    let iface = Marshal.from_channel chan in
+    let _ = close_in chan in
+    if iface.hash == Digest.file src_path then
```

Comment:
This line is not correct---the imported library might have been changed as well. That is,

unit a -> unit b -> unit c

There might be indirect dependencies.

Reply:
Nice catch! Will factor this in to the iface design.